argus/action_manager/windows.py

```diff
@@ -171,20 +172,41 @@ def sysprep(self):
         LOG.info("Wait for the machine to finish rebooting ...")
         self.wait_boot_completion()
 
-    def git_clone(self, repo_url, location):
-        """Clone from an remote repo to a specific location on the instance.
-
-        :param repo_url:
-            The remote repo url.
-        :param location:
-            Specific location on the instance.
+    def git_clone(self, repo_url, location, count=util.RETRY_COUNT,
+                  delay=util.RETRY_DELAY):
+        """Clone from a remote repository to a specified location.
+
+        :param repo_url: The remote repository url.
+        :param location: The target location for where to clone the repository.
+        :param count:
+            The number of tries that should be attempted in case it fails.
+        :param delay: The time delay before retrying.
+        :returns: True if the clone was successful, False if not.
+        :raises: ArgusCLIError if the path is not valid.
+        :rtype: bool
         """
+        if self.exists(location):
```

Comment:
The `exists()` method could raise `ArgusTimeoutError` exception. I think, you should include this statement in a `try-except` block and `return False` if it could not check the existence of `location`.